### PR TITLE
updated sendmail action

### PR DESCRIPTION
**Pull request by NiekD**
Added validity check for mail address
Adapted to new library structure
Assigned action arguments to mail struct directly
Several fixes

### CODE REVIEW

File: libs/pilight/events/actions/sendmail.c [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+#if !defined(MODULE) && !defined(_WIN32)
+__attribute__((weak))
+#endif
+void actionSendmailInit(void) {
+	event_action_register(&action_sendmail, "sendmail");
+
+	options_add(&action_sendmail->options, 'a', "SUBJECT", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
+	options_add(&action_sendmail->options, 'b', "MESSAGE", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
+	options_add(&action_sendmail->options, 'c', "TO", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
```
Comment (CurlyMoo) [inline libs/pilight/events/actions/sendmail.c]: Didn't the mail regex work here?

Comment (NiekD) [inline libs/pilight/events/actions/sendmail.c]: No, indeed it didn't work here, so I did it in the arguments check.

CurlyMoo schreef op 2-4-2015 om 21:17:

**Comment by CurlyMoo** [comment]
Thanks

**Comment by NiekD** [comment]
Shall I add a description to the action wiki?


CurlyMoo schreef op 3-4-2015 om 07:01:

**Comment by CurlyMoo** [comment]
Yes always :+1:

**Comment by NiekD** [comment]
Done

CurlyMoo schreef op 3-4-2015 om 09:16:

**Pull request merged**